### PR TITLE
Added isUsed property for Attribute Output node

### PR DESCRIPTION
**Pull request by harisreedhar**

### CODE REVIEW

**Comment by OmarEmaraDev** [comment]
What is the rational behind the change?

By the way, don't commit your changes to master. Create a new branch and commit it there. Your master should be identical to upstream master.

**Comment by harisreedhar** [comment]
> What is the rational behind the change?

Sometimes we only want to set attribute only once, not every time. which saves some performance.
For example setting object's display type to bounds, which doesn't requires to execute every time. but only once.
There are many other cases as well like setting rigidbody mass, collision shape etc.

**Comment by harisreedhar** [comment]
> By the way, don't commit your changes to master. Create a new branch and commit it there. Your master should be identical to upstream master.

Ok i will take care of that, do i need to create new pull request from new branch?

File: CHANGELOG.md [hunk 1]
```diff
@@ -59,6 +59,7 @@
 - Remove clamping from the inputs of the *Random Falloff* node.
 - Support FLIP particle systems in the *Particles Data* node.
 - Vectorized *Point Distance Falloff* node.
+- Added *Attribute Output* node enable/disable property for value socket.
```
Comment (OmarEmaraDev) [inline CHANGELOG.md:62]: I think a clearer statement would be:

```
Added *Is Used* property to the value socket in the *Attribute Output* node.
```

File: animation_nodes/nodes/object/object_attribute_output.py [hunk 2]
```diff
@@ -44,28 +46,29 @@ def getExecutionCode(self, required):
             yield "self.setErrorMessage('Invalid Syntax', show = len(self.attribute.strip()) > 0)"
             return
 
-        yield "try:"
-        if self.useObjectList:
-            if self.useValueList:
-                yield "    _values = [None] if len(values) == 0 else values"
-                yield "    _values = itertools.cycle(_values)"
-                yield "    for object, value in zip(objects, _values):"
+        if self.inputs[1].isUsed:
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/object/object_attribute_output.py:49]: Use a guard condition here.
```
if not self.inputs[1].isUsed: return
```

**Review by OmarEmaraDev — CHANGES_REQUESTED** [review]
Okay. Sounds reasonable.

You don't have to recreate the pull request or the branch now, just make sure to do it in the future.

File: CHANGELOG.md (hunk 1 above)
Comment (harisreedhar) [reply CHANGELOG.md]: ok

**Comment by harisreedhar** [comment]
@OmarEmaraDev Do i need to add a release note?

**Review by OmarEmaraDev — APPROVED** [review]
Thanks!

Yes, just add an entry like this https://docs.animation-nodes.com/release_notes/v2_2/object/.

**Pull request merged**